### PR TITLE
Decouple signature validation from state machine

### LINKED ISSUE #3858: Move signature packing outside the state machine
The reason we want to do this is because:
1. Be able to reject messages with invalid signatures early.
2. Enable the tests to provide arbitrary signatures and remove the boilerplate code which requires all the signature generation.

## TODO
- [ ] move the [balance_proof.py](https://github.com/raiden-network/raiden/blob/master/raiden/transfer/balance_proof.py) module outside the state-machine codebase
- [ ] Update callers
- [ ] Verify balance-proof containing messages before the state change is dispatched into the WAL and potentially reject the message if the signature doesn't match.

### PR DESCRIPTION
**Pull request by rakanalh**
Resolves #3858

### CODE REVIEW

File: raiden/balance_proof.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+            channel_identifier=envelope_message.channel_identifier,
+        ),
+    )
+
+    signature_is_valid = is_valid_signature(
+        balance_proof=balance_proof,
+        sender_address=envelope_message.sender,
+    )
+
+    if signature_is_valid:
```
Comment (pirapira) [inline raiden/balance_proof.py]: I think this condition always holds.

```
if (False, 'msg'):
    print(3)
else:
    print(4)
```
shows me 3.

**Comment by pirapira** [comment]
Also add a test that calls `balanceproof_from_envelope()` with some broken signature, and expects `None`.
